utils/python/CIME/create_test.py

```diff
@@ -95,17 +95,37 @@ def __init__(self, test_names,
         else:
             self._parallel_jobs = parallel_jobs
 
-        if (self._compare or self._generate):
-
+        self._baseline_cmp_name = None
+        self._baseline_gen_name = None
+        self._compare = False
+        self._generate = False
+        if (compare or generate):
+            logging.info("compare is %s generate is %s"%(compare,generate))
             # Figure out what baseline name to use
             if (baseline_name is None):
+                if(compare is not None and compare == str(compare)):
```

Comment:
You should use the isinstance function: if isinstance(compare, str)

On Thu, Feb 11, 2016 at 4:09 PM, James Foucar <notifications@github.com>
wrote:

> In utils/python/CIME/create_test.py
> <https://github.com/ESMCI/cime/pull/60#discussion_r52683220>:
>
> >              # Figure out what baseline name to use
> >              if (baseline_name is None):
> > +                if(compare is not None and compare == str(compare)):
>
> To check if compare is a string, you want this: "type(compare) is str"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ESMCI/cime/pull/60/files#r52683220>.
>